code/part-one/blockchain.js

```diff
@@ -15,69 +31,93 @@ class Transaction {
+ */
 class Block {
-  /*
-    Initialize the block constructor
-
-    **Note**
-    The block should be initiated with
-      - a timestamp
-      - transactions
-      - the previous block's hash
-  */
+  /**
+   * Accepts an array of transactions and the hash of a previous block. It
+   * saves these and uses them to calculate a hash.
+   *
+   * Properties:
+   *   - transactions: the passed in transactions
+   *   - previousHash: the passed in hash
+   *   - nonce: just set this to some hard-coded number for now, it will be
+   *     used later when we make blocks mineable with our own PoW algorithm
+   *   - hash: a unique hash string generated from the other properties
+   */
   constructor(transactions, previousHash) {
     this.transactions = transactions;
     this.previousHash = previousHash;
     this.calculateHash(0);
   }
 
-  /*
-    Calculate a hash based on transactions supplied to block:
-
-    - Add functionality to return a hash by passing a compiled string
-    into the SHA256 hashing function
-
-    - The string that is passed in should contain specific data
-    about the block that would be impossible to replicate
-
-    **Hint**
-    Unreplicable data can include
-      - transactions
-      - the previous hash
-      - the block timestamp
-  */
+  /**
+   * Accepts a nonce, and generates a unique hash for the block. Updates the
+   * hash and nonce properties of the block accordingly.
+   *
+   * Hint:
+   *   The format of the hash is up to you. Remember that it needs to be
+   *   unique and deterministic, and must become invalid if any of the block's
+   *   properties change.
+   */
   calculateHash(nonce) {
     const transactionString = this.transactions.map(t => t.signature).join('');
     const toHash = this.previousHash + transactionString + nonce;
+
     this.nonce = nonce;
     this.hash = createHash('sha512').update(toHash).digest('hex');
   }
 }
 
+/**
+ * A Blockchain class for storing an array of blocks, each of which is linked
+ * to the previous block by their hashes. Includes methods for adding blocks,
+ * fetching the head block, and checking the balances of public keys.
+ */
 class Blockchain {
+  /**
+   * Generates a new blockchain with a single genesis block. A genesis block
+   * is important so later blocks have something to build off of. It should
+   * have no transactions, and `null` for a previous hash.
+   *
+   * Properties:
+   *   - blocks: an array of blocks, starts with one genesis block
+   */
   constructor() {
     const genesis = new Block([], null);
     this.blocks = [ genesis ];
   }
 
-  /*
-    Get the last created block:
-
-    - Add functionality to get the latest block in the chain
-  */
+  /**
+   * Simply returns the last block added to the chain.
+   */
   getHeadBlock() {
     return this.blocks[this.blocks.length - 1];
   }
 
+  /**
+   * Accepts an array of transactions and adds them to the chain in
+   * a new block.
+   *
+   * Hint:
+   *   Don't forget the hashes!
```

Comment:
Hmm, I think we should see how students respond to this hint! (I'm wonder if it's too vague..)